extractor.php

```diff
@@ -88,37 +96,55 @@
 
         $html = new Html();
         $html->retrieve($alert_url->url);
+
+        if(!$html->exist)
+        {
+            $critical_error = true;
+            goto log_critical;
+        }
+
         $ads = $html->parse($source->id);
         
         /**
          * Analyze the parsed ads and insert or
          * update if any change has been detected
          * or there's a new ad found.
          * 
+         * It will analyze the vehicle color if necessary.
+         * 
          */
         foreach($ads as $ad)
         {
+            $logger->ads_found++;
             if($found_ad = Advertisement::findUnique($ad->source_id.$ad->reference))
             {
                 $ad->id = $found_ad->id;
                 if($ad != $found_ad)
                 {
-                    $ad->save();
+                    if($ad->save())
+                    {
+                        $logger->ads_updated++;
+                    }else{
+                        $logger->errors_found++;
+                    }
```

Comment:
if/else block has wrong indentation
```suggestion
                    }else
                    {
                        $logger->errors_found++;
                    }
```